notebook/notebookapp.py

```diff
@@ -1100,6 +1109,13 @@ def start(self):
         
         This method takes no arguments so all configuration and initialization
         must be done prior to calling this method."""
+        try:
+            if os.geteuid() == 0 and not self.allow_root:
+                self.log.critical("Running as root is forbidden. Use --allow-root to bypass.")
+                self.exit(1)
+        except OSError as e:
+            pass
```

Comment:
@Carreau What docs are those? I rebooted into Windows to check, and it ain't there:

```
>>> os.geteuid()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'os' has no attribute 'geteuid'
```